astarte-setup.sh

```diff
@@ -0,0 +1,269 @@
+            ;;
+        -y)
+            ASTARTE_SETUP_NON_INTERACTIVE=true
+            shift 1
+            ;;
+        --help)
+            print_help
+            exit 0
+            ;;
+        *)
+            echo "Invalid argument: $1"
+            print_help
+            exit 1
+            ;;
+    esac
+done
+
+if [ ! -d "$ASTARTE_SETUP_CLONE_DIR" ]; then
+    echo "Provided clone directory does not exists: $ASTARTE_SETUP_CLONE_DIR."
+    exit 1
+fi
+ASTARTE_SETUP_CLONE_DIR="$ASTARTE_SETUP_CLONE_DIR/astarte"
+
+if [ ! -d "$ASTARTE_SETUP_CLONE_DIR" ]; then
+    mkdir "$ASTARTE_SETUP_CLONE_DIR"
+else
+    # The clone directory is not empty
+    if [ $(ls -A "$ASTARTE_SETUP_CLONE_DIR" | wc -l) -gt 0 ]; then
+        echo "Provided clone directory directory already contains a non empty astarte folder."
+        exit 1
+    fi
+fi
+
+# Ask for y|N
+prompt() {
+    read -rp "$1 (y/N) " p
+    if [ "$p" == "y" ]; then
+        p="y"
+    else
+        p="n"
+    fi
+}
+
+# Prompt and then run the optional command
+ask() {
+    if $ASTARTE_SETUP_NON_INTERACTIVE; then
+        $2
+        return
+    fi
+    prompt "$1"
+    if [ "$p" == "y" ]; then
+        $2
+    fi
+}
+
+# Prompt and then run the required command, or fail setup procedure
+hard_ask() {
+    if $ASTARTE_SETUP_NON_INTERACTIVE; then
+        $2
+        return
+    fi
+    prompt "$1"
+    if [ "$p" == "y" ]; then
+        $2
+    else
+        echo "Astarte setup aborted."
+        exit 1
+    fi
+}
+
+tell_and_wait() {
+  echo "$1"
+    if $ASTARTE_SETUP_NON_INTERACTIVE; then
+        return
+    fi
+  read -p "Press Enter to continue..."
+}
+
+# Check the aio-max-nr for Scylla
+check_aio_max() {
+    aio_max_nr="$(sysctl -n fs.aio-max-nr || cat /proc/sys/fs/aio-max-nr)"
+    if [[ $aio_max_nr < 1048576 ]]; then
+        echo "You need to update the sys 'fs.aio-max-nr' to be atleast 1048576 to run ScyllaDB"
+        echo "You can use the command:"
+        echo "    sudo sysctl -w fs.aio-max-nr=1048576"
+        echo
+        exit 1
+    fi
+}
+
+find_docker_compose() {
+    if [[ -n ${DOCKER_COMPOSE:-} ]]; then
+        echo "Using the '$DOCKER_COMPOSE' command" >&2
+        return
+    elif docker compose version; then
+        export DOCKER_COMPOSE='docker compose'
+    elif docker-compose version; then
+        export DOCKER_COMPOSE='docker-compose'
+    else
+        echo "Couldn't run the 'docker-compose' nor 'docker compose' commands"
+        echo "You can specify a by setting the DOCKER_COMPOSE env variable"
+        exit 1
+    fi
+}
+
+find_astartectl() {
+    if ! command -v astartectl; then
+        echo "Cannot find 'astartectl' in PATH" >&2
+        echo "Please download and install the latest release for your system" >&2
+        echo "You can find it here: https://github.com/astarte-platform/astartectl" >&2
+        exit 1
+    fi
+}
+
+clone_astarte() {
+    git clone https://github.com/astarte-platform/astarte.git -b v1.2.0 "$ASTARTE_SETUP_CLONE_DIR"
```

Comment:
What do you think moving Astarte version to a const variable, so that when updating the Astarte version, you only have a single point ?
```suggestion
    git clone https://github.com/astarte-platform/astarte.git -b $ASTARTE_VERSION "$ASTARTE_SETUP_CLONE_DIR"
```